features/messages/impl/src/test/kotlin/io/element/android/features/messages/timeline/factories/event/TimelineItemContentPollFactoryTest.kt

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.timeline.factories.event
+
+import com.google.common.truth.Truth
+import io.element.android.features.messages.impl.timeline.factories.event.TimelineItemContentPollFactory
+import io.element.android.features.messages.impl.timeline.model.event.TimelineItemPollContent
+import io.element.android.features.poll.api.PollAnswerItem
+import io.element.android.libraries.featureflag.api.FeatureFlags
+import io.element.android.libraries.featureflag.test.FakeFeatureFlagService
+import io.element.android.libraries.matrix.api.poll.PollAnswer
+import io.element.android.libraries.matrix.api.poll.PollKind
+import io.element.android.libraries.matrix.api.timeline.item.event.PollContent
+import io.element.android.libraries.matrix.test.A_USER_ID
+import io.element.android.libraries.matrix.test.A_USER_ID_10
+import io.element.android.libraries.matrix.test.A_USER_ID_2
+import io.element.android.libraries.matrix.test.A_USER_ID_3
+import io.element.android.libraries.matrix.test.A_USER_ID_4
+import io.element.android.libraries.matrix.test.A_USER_ID_5
+import io.element.android.libraries.matrix.test.A_USER_ID_6
+import io.element.android.libraries.matrix.test.A_USER_ID_7
+import io.element.android.libraries.matrix.test.A_USER_ID_8
+import io.element.android.libraries.matrix.test.A_USER_ID_9
+import io.element.android.libraries.matrix.test.FakeMatrixClient
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+private const val A_POLL_QUESTION = "What is your favorite food?"
+private val A_POLL_ANSWER_1 = PollAnswer("id_1", "Pizza")
+private val A_POLL_ANSWER_2 = PollAnswer("id_2", "Pasta")
+private val A_POLL_ANSWER_3 = PollAnswer("id_3", "French Fries")
+private val A_POLL_ANSWER_4 = PollAnswer("id_4", "Hamburger")
+
+internal class TimelineItemContentPollFactoryTest {
+
+    private val factory = TimelineItemContentPollFactory(
+        matrixClient = FakeMatrixClient(),
+        featureFlagService = FakeFeatureFlagService(mapOf(FeatureFlags.Polls.key to true)),
+    )
+
+    @Test
+    fun `Disclosed poll - not ended states`() = runTest {
+        // No votes
+        Truth.assertThat(
+            factory.create(aPollContent(PollKind.Disclosed))
+        ).isEqualTo(aTimelineItemPollContent(PollKind.Disclosed))
+
+        // Some votes, according one from current user
```

Comment:
```suggestion
        // Some votes, including one from current user
```

There are other occurrences to fix in the file.